ark/phenotyping/run_cell_som.R

```diff
@@ -16,44 +17,48 @@ args <- commandArgs(trailingOnly=TRUE)
 # get the path to the cluster counts data
 clusterCountsPath <- args[1]
 
+# get the path to the cluster counts norm data
+clusterCountsNormPath <- args[2]
+
 # get the weights write path
-cellWeightsPath <- args[2]
+cellWeightsPath <- args[3]
 
 # get the cluster write path
-cellClusterPath <- args[3]
+cellClusterPath <- args[4]
 
-# read the cluster counts data
+# read the cluster counts data (norm and un-norm)
 print("Reading the cluster counts data")
 clusterCountsData <- arrow::read_feather(clusterCountsPath)
+clusterCountsNorm <- arrow::read_feather(clusterCountsNormPath)
 
 # read the weights
 print("Reading the weights matrix")
 somWeights <- as.matrix(arrow::read_feather(cellWeightsPath))
 
-# get the column names of the SOM pixel/meta clusters
-clusterCols <- colnames(clusterCountsData)[grepl("cluster_|hCluster_cap_",
-                                           colnames(clusterCountsData))]
+clusterCols <- colnames(clusterCountsNorm)[grepl("cluster_|hCluster_cap_",
+                                           colnames(clusterCountsNorm))]
 
 # keep just the cluster columns
-clusterCountsNorm <- clusterCountsData[, clusterCols]
+clusterCountsNorm <- as.matrix(clusterCountsNorm[,clusterCols])
 
 # 99.9% normalize
 print("Perform 99.9% normalization")
 for (clusterCol in clusterCols) {
-    normVal <- quantile(clusterCountsNorm[,clusterCol])
+    normVal <- quantile(clusterCountsNorm[,clusterCol], 0.999)
 
-    # prevent normalizing by 0
-    if (normVal != 0) {
-        clusterCountsNorm[,clusterCol] <- clusterCountsNorm[,clusterCol] / normVal
+    if (normVal == 0) {
+        normVal <- quantile(clusterCountsNorm[,clusterCol], 1)
     }
+
+    clusterCountsNorm[,clusterCol] <- clusterCountsNorm[,clusterCol] / normVal
 }
 
 # map FlowSOM data
 print("Mapping cluster labels")
 clusters <- FlowSOM:::MapDataToCodes(somWeights, as.matrix(clusterCountsNorm))
 
 # assign cluster labels to pixel data
-clusterCountsData$cluster <- clusters[,1]
+clusterCountsData$cluster <- as.integer(clusters[,1])
```

Comment:
I might be missing something here, but why are you appending the cluster numbers to clusterCountsData and not clusterCountsNorm? Right now you are saving clusterCountsData and then using that to calculate the averages, which is then fed into consensus clustering. But consensus clustering should be running on the size-normalized data.

Reply:
Oh great catch! I think at one point I had to save the normalized and the un-normalized versions with the now-extraneous function calls, but I never got around to saving the normalized version.